src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Services/DataSetVersionService.cs

```diff
@@ -23,31 +23,50 @@ internal class DataSetVersionService(
+                releaseFileId: releaseFileId,
+                instanceId: instanceId,
+                dataSet: dataSet, 
+                cancellationToken: cancellationToken));
+    }
+
+    public async Task<Either<ActionResult, (Guid dataSetId, Guid dataSetVersionId)>> CreateNextVersion(
+        Guid dataSetId,
+        Guid releaseFileId,
+        Guid instanceId,
+        CancellationToken cancellationToken = default)
+    {
+        return await GetDataSet(dataSetId, cancellationToken)
+            .OnSuccess(ValidateNextDataSet)
+            .OnSuccess(dataSet => CreateDataSetVersion(
+                releaseFileId: releaseFileId,
+                instanceId: instanceId,
+                dataSet: dataSet, 
+                cancellationToken: cancellationToken));
+    }
+    
     public async Task<Either<ActionResult, Unit>> DeleteVersion(Guid dataSetVersionId,
         CancellationToken cancellationToken = default)
     {
-        var strategy = contentDbContext.Database.CreateExecutionStrategy();
-
-        return await strategy.ExecuteAsync(
-            async () =>
-            {
-                using var transactionScope = new TransactionScope(
-                    TransactionScopeOption.Required,
-                    new TransactionOptions { IsolationLevel = IsolationLevel.ReadCommitted },
-                    TransactionScopeAsyncFlowOption.Enabled);
-
-                return await GetDataSetVersion(dataSetVersionId, cancellationToken)
-                    .OnSuccessDo(CheckCanDeleteDataSetVersion)
-                    .OnSuccessDo(async dataSetVersion => await GetReleaseFile(dataSetVersion, cancellationToken)
-                        .OnSuccessVoid(async releaseFile =>
-                            await UpdateFilePublicApiDataSetId(releaseFile, cancellationToken))
-                        .OnFailureDo(_ =>
-                            throw new KeyNotFoundException(
-                                $"The expected 'ReleaseFile', with ID '{dataSetVersion.ReleaseFileId}', was not found.")))
-                    .OnSuccessDo(async dataSetVersion => await DeleteDataSetVersion(dataSetVersion, cancellationToken))
-                    .OnSuccessVoid(DeleteParquetFiles)
-                    .OnSuccessVoid(transactionScope.Complete);
-            });
+        return await publicDataDbContext.RequireTransaction(() => 
+            GetDataSetVersion(dataSetVersionId, cancellationToken)
+                .OnSuccessDo(CheckCanDeleteDataSetVersion)
+                .OnSuccessDo(async dataSetVersion => await GetReleaseFile(dataSetVersion, cancellationToken)
+                    .OnSuccessVoid(async releaseFile =>
+                        await UpdateFilePublicApiDataSetId(releaseFile, cancellationToken))
+                    .OnFailureDo(_ =>
+                        throw new KeyNotFoundException(
+                            $"The expected 'ReleaseFile', with ID '{dataSetVersion.ReleaseFileId}', was not found.")))
```

Comment:
This has been changed on dev - need to rebase.

Reply:
Seen that during rebase, thanks.  Updated